### PR TITLE
Added a not found view to product grid

### PR DESCRIPTION
**Pull request by mikemurray**
- Added new component NotFound that lets you do blank views
- Added not found view to product grid if tag was not found

### CODE REVIEW

**Comment by zenweasel** [comment]
@mikemurray If I go to a non-existent tag I don't get a 404 I get a "No Products Found". Is that the correct behavior?

**Comment by zenweasel** [comment]
Is it my imagination or does rendering of products now seem considerably slower?

**Comment by zenweasel** [comment]
If I go to a tag with no products I just get nothing rather than the "No Products Found" page.

**Comment by mikemurray** [comment]
I think I only did, `if the tag does not exist, then no products found`; which is more like a tag not found. Should extend this to, if the products array is empty, then no product found as well.

---

The load may be slower / feel slower.

**Comment by zenweasel** [comment]
I feel like if I go to a tag with no products I should get the "No product" message rather than just a blank page which looks broken. What do you think?

That's the way it currently works on `master` (blank page) so it's not a bug/blocker

**Comment by mikemurray** [comment]
I agree, would be easy to add I think.

File: imports/plugins/core/ui/client/components/notFound/notFound.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ */
+const NotFound = (props) => {
+  // ClassName for base wrapper,
+  // If one is provied in props, the default is not used
+  const baseClassName = classnames({
+    "container-fluid-sm": typeof props.className === "undefined"
+  }, props.className);
+
+  // ClassName for content wrapper
+  // If one is provied in props, the default is not used
```
Comment (zenweasel) [inline imports/plugins/core/ui/client/components/notFound/notFound.js]: typo in comment

File: imports/plugins/core/ui/client/components/notFound/notFound.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+NotFound.propTypes = {
+  /**
+   * Children
+   * @type {ReactNode}
+   */
+  children: PropTypes.node,
+
+  /**
+   * Baase wrapper className
+   * @summary String className of `classnames` compatable object for the base wrapper
```
Comment (zenweasel) [inline imports/plugins/core/ui/client/components/notFound/notFound.js]: typo in comment

File: imports/plugins/core/ui/client/components/notFound/notFound.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        {props.children}
+      </div>
+    </div>
+  );
+};
+
+NotFound.propTypes = {
+  /**
+   * Children
+   * @type {ReactNode}
```
Comment (zenweasel) [inline imports/plugins/core/ui/client/components/notFound/notFound.js]: Should we make this more consistent? I feel like we call this different things different place (e.g. Node/ReactNode/JSX. Is there a difference?)

File: imports/plugins/core/ui/client/components/notFound/notFound.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+   */
+  children: PropTypes.node,
+
+  /**
+   * Baase wrapper className
```
Comment (zenweasel) [inline imports/plugins/core/ui/client/components/notFound/notFound.js]: typo in comment

File: imports/plugins/core/ui/client/components/notFound/notFound.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+   * @summary String className of `classnames` compatable object for the base wrapper
+   * @type {String|Object}
+   */
+  className: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+
+  /**
+   * Content className
+   * @summary String className of `classnames` compatable object for the content wrapper
```
Comment (zenweasel) [inline imports/plugins/core/ui/client/components/notFound/notFound.js]: typo in comment (compatible)

File: imports/plugins/core/ui/client/components/notFound/notFound.js (hunk 3 above)
Comment (mikemurray) [inline imports/plugins/core/ui/client/components/notFound/notFound.js]: Maybe just go with `Node` prop type.

@machikoyasuda what do you think? Maybe we can set up a type in jsdoc that knows `Node` is react node when the docs are generated?

**Comment by mikemurray** [comment]
@zenweasel Should now show the not-found view if the tag is valid, but no products were found still.

@machikoyasuda As a bonus, jsdoc'd all the things in `Products` component.

File: imports/plugins/core/ui/client/components/notFound/notFound.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+NotFound.propTypes = {
+  /**
+   * Children
+   * @type {Node}
+   */
+  children: PropTypes.node,
+
+  /**
+   * Base wrapper className
+   * @summary String className or `classnames` compatable object for the base wrapper
```
Comment (zenweasel) [inline imports/plugins/core/ui/client/components/notFound/notFound.js]: compatible still spelled wrong isn't it?

**Comment by zenweasel** [comment]
@mikemurray three lint errors that seem to come from nowhere

**Comment by mikemurray** [comment]
@zenweasel wonder if something changed in our eslint config since those files were passing before, right?

**Comment by mikemurray** [comment]
Fixed all lint issues

**Comment by zenweasel** [comment]
@mikemurray yeah, I don't know where those lint errors came from but thanks for fixing them

**Comment by zenweasel** [comment]
@mikemurray @machikoyasuda still two lint errors left

**Comment by mikemurray** [comment]
ok, whats going on. I fixed all the lint errors and more keep poping up?

**Comment by mikemurray** [comment]
All green!

**Pull request merged**